### PR TITLE
User can buy a subscription

### PR DESCRIPTION
**Pull request by EevanR**
[PT-Link](https://www.pivotaltracker.com/story/show/170646669)

- adds subscriptions controller with create action
- adds stripe gem and stripe-ruby-mock gem
- add subscribers role
- add subscription create route
- remedy namespace issue

### CODE REVIEW

File: spec/requests/api/v1/subscription_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    params: {
+    stripeToken: 123456789
+    }, headers: headers}
+
+  describe "User pays for subscription" do
+    before do
+      successful_token
+    end
+
+    it 'successfully' do
```
Comment (oliverochman) [inline spec/requests/api/v1/subscription_spec.rb]: please redo the structure of this test

```
RSpec.describe 'User can buy subscritption' do
  describe 'successfully' do
  end
  describe 'unsuccessfully' do
    describe 'with invalid stripe token' do
    end
  end
end

```

File: spec/requests/api/v1/subscription_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  user.reload }
+
+  let(:invalid_token) { post '/api/v1/subscriptions', 
+    params: {
+    stripeToken: 123456789
+    }, headers: headers}
+
+  describe "User pays for subscription" do
+    before do
+      successful_token
```
Comment (oliverochman) [inline spec/requests/api/v1/subscription_spec.rb]: I see no reason to extract the request to a variable, you guys have not done it previously. Keep it consistent.

File: spec/requests/api/v1/subscription_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      expect(user.role).to eq 'subscriber'
+    end
+  end
+
+  describe "User pays for subscription with invalid token" do
+    before do
+      invalid_token
+    end
+
+    it 'recieves error message' do
```
Comment (oliverochman) [inline spec/requests/api/v1/subscription_spec.rb]: are there no other sad paths then this one?

File: config/credentials.yml.enc [hunk 4]
```diff
@@ -1 +1 @@
-cTlXKCqJcBVl733P+/yb1LXqTgvV/y0D2LtFTePTIdV4J0G6p0BOAYrKLjUjaTvtDEMic6Yc6vEwHcAumAg3EYDfF4YbtDh5cigKIqm578zZwu8zQ38XAib3paWrCBaRcCi6NssjyMhF7xSrgP6uCYZ/t3kvFgYBNR6f+nHPt9/eWVyGkHPNobE2wiyzWjNYxzwLkxz55xt7Wa7Gr/vGCVNSZdlf0XL5qPt7oddqcdPaGwC8hKwPQ/3YLOlTsx/RMFy5ypaeRxM4Q9bVGO2iuqm4QePeb2OPSxzg+XgnBe9M71rA+0rYASNcdh9fx4AO+wbPQWu/RIoEIQF/gcoLWHOg2hCRwGPzFmuuO2WRsFMbDS7H1R/IheuEyI57ve2ZFfZ0MoizcSL/3LO8bLRJPc8F/EFRywmcqAWg--4PeiqEcd9kGoDtYl--xGfnseXkWayyt2FTUTBrRQ==
\ No newline at end of file
+XWOb4rEq4VTHwKy7BaInm+Vp9gkXkkpUJfz5douUIueJkjImwqEc+4g+1mDU1xJGYGPPdXLh+T8szevt8ReTjCeqZUqqEypgkPqsgVkGnTfAn69aVMK3Aj3mwGp/FsjS+4ZtcLcuEf5rizCNMSUfiFFC7TTLn/eIME9LR82mVOZNvRIEFsD/0CAq1iORi6Hr/1A7sM/pZmdyZY+WemMh7YWzrL1/anUAdZRAUzPYZpqJZhKATRukn8hvnmICzxCXSe4qxry0l7FFuH7gpR21Rbcdzd+1uJeZ7BRCcG5P6f9bUMar3AwBY32F2asLoYBJosdayKLagFYlB00e0Wd4DDWovVd9zb1LlkTdqZicP575Tu/QQNqJZq8Avic4kUpQkgDnw8bRPxzwDe3tzXwANQMr1UwwonGzYbf32Avf1TqDYj6xF+EpqK5sxIvQRoqRzQNeSrrR45nf0VjSdJuoISrQVvl8UqyMb/JywSMt3DkVuSB034ZwE8hwfAm9drVt22NY0hU8m1jfn19CHjrBGRXxN/bRB1Wf8xH7NpRTCA==--1JOnAEuLLhyxAheD--vocyfqHYC9OhayyxYcwNlQ==
```
Comment (oliverochman) [inline config/credentials.yml.enc:1]: why have the credentials changed?
did you generate a new master.key?

File: app/controllers/api/v1/subscriptions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,31 @@
+class Api::V1::SubscriptionsController < ApplicationController
+  before_action :authenticate_user!
+  rescue_from Stripe::InvalidRequestError, with: :invalid_token
```
Comment (oliverochman) [inline app/controllers/api/v1/subscriptions_controller.rb]: there must be other potential errors that can happen here?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/subscription_spec.rb (hunk 1 above)
Comment (EevanR) [reply spec/requests/api/v1/subscription_spec.rb]: ok, made it a bit more descriptive

File: spec/requests/api/v1/subscription_spec.rb (hunk 2 above)
Comment (EevanR) [reply spec/requests/api/v1/subscription_spec.rb]: right you are oliver! i put the request in the before do block

File: config/credentials.yml.enc (hunk 4 above)
Comment (EevanR) [reply config/credentials.yml.enc:1]: we put the stripe keys from thomas

File: app/controllers/api/v1/subscriptions_controller.rb (hunk 5 above)
Comment (EevanR) [reply app/controllers/api/v1/subscriptions_controller.rb]: added another potential error

File: spec/requests/api/v1/subscription_spec.rb (hunk 3 above)
Comment (EevanR) [reply spec/requests/api/v1/subscription_spec.rb]: added another sad path

**Review by oliverochman — APPROVED** [review]

**Pull request merged**